### PR TITLE
Update iOS 4.10.0 (and add AdServices)

### PR DESCRIPTION
**Pull request by joshdholtz**
- Update iOS to 4.10.0 with AdServices
  - New public method `enableAdServicesAttributionTokenCollection`

### CODE REVIEW

File: ios/PurchasesHybridCommon/PurchasesHybridCommon/CommonFunctionality.swift [hunk 1]
```diff
@@ -63,7 +63,15 @@ import RevenueCat
     }
 
     @objc public static func setAutomaticAppleSearchAdsAttributionCollection(_ enabled: Bool) {
-        Purchases.automaticAppleSearchAdsAttributionCollection = enabled
+        // Using `setValue` because `automaticAppleSearchAdsAttributionCollection` is now deprecated and we get a warning
+        Purchases.setValue(enabled, forKey: "automaticAppleSearchAdsAttributionCollection")
```
Comment (taquitos) [inline ios/PurchasesHybridCommon/PurchasesHybridCommon/CommonFunctionality.swift:67]: CatBobbing

**Review by taquitos — APPROVED** [review]
🎈🐐

File: ios/PurchasesHybridCommon/PurchasesHybridCommon/CommonFunctionality.swift (hunk 1 above)
Comment (NachoSoto) [reply ios/PurchasesHybridCommon/PurchasesHybridCommon/CommonFunctionality.swift:67]: ⚠️  this seems super fragile.
Why don't we mark `setAutomaticAppleSearchAdsAttributionCollection` as deprecated as well? That way we don't need the hack, and we propagate the deprecation.

Comment (NachoSoto) [reply ios/PurchasesHybridCommon/PurchasesHybridCommon/CommonFunctionality.swift:67]: #199

**Pull request merged**